base/linalg/triangular.jl

```diff
@@ -1827,8 +1827,26 @@ function eigvecs{T}(A::AbstractTriangular{T})
 end
 det{T}(A::UnitUpperTriangular{T}) = one(T)
 det{T}(A::UnitLowerTriangular{T}) = one(T)
+logdet{T}(A::UnitUpperTriangular{T}) = zero(T)
+logdet{T}(A::UnitLowerTriangular{T}) = zero(T)
+logabsdet{T}(A::UnitUpperTriangular{T}) = zero(T), one(T)
+logabsdet{T}(A::UnitLowerTriangular{T}) = zero(T), one(T)
 det{T}(A::UpperTriangular{T}) = prod(diag(A.data))
 det{T}(A::LowerTriangular{T}) = prod(diag(A.data))
+function logabsdet{T}(A::Union{UpperTriangular{T},LowerTriangular{T}})
+    sgn = one(T)
+    absdet = zero(real(T))
+    @inbounds for i in 1:size(A,1)
+        diag_i = A.data[i,i]
+        sgn *= sign(diag_i)
+        abs_det += log(abs(diag_i))
+    end
+    return abs_det, sgn
+end
+function logdet{T}(A::Union{UpperTriangular{T},LowerTriangular{T}})
```

Comment:
No, but this implementation should be valid for any `AbstractMatrix` type that implements `logabsdet`. So probably good to replace this line https://github.com/JuliaLang/julia/blob/758aebca25fbe8a94c57c251a285f54e6e45b0bf/base/linalg/generic.jl#L538 with something that delegates to `logabsdet` the way you have it here, instead of always doing `lufact`.